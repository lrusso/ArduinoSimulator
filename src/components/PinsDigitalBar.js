import React from "react"
import PinsDigitalLabel from "./PinsDigitalLabel"
import PinsDigitalItem from "./PinsDigitalItem"
import { useSimulatorContext } from "../contexts/SimulatorContext"

const PinsDigitalBar = () => {
  const {
    digitalPin0,
    digitalPin1,
    digitalPin2,
    digitalPin3,
    digitalPin4,
    digitalPin5,
    digitalPin6,
    digitalPin7,
    digitalPin8,
    digitalPin9,
    digitalPin10,
    digitalPin11,
    digitalPin12,
    digitalPin13,
  } = useSimulatorContext()

  return (
    <div style={styles.container}>
      <div style={styles.noScrollbar}>
        <div style={styles.wrapper}>
          <PinsDigitalLabel />
          <PinsDigitalItem pinNumber={0} isEnabled={digitalPin0} />
          <PinsDigitalItem pinNumber={1} isEnabled={digitalPin1} />
          <PinsDigitalItem pinNumber={2} isEnabled={digitalPin2} />
          <PinsDigitalItem pinNumber={3} isEnabled={digitalPin3} />
          <PinsDigitalItem pinNumber={4} isEnabled={digitalPin4} />
          <PinsDigitalItem pinNumber={5} isEnabled={digitalPin5} />
          <PinsDigitalItem pinNumber={6} isEnabled={digitalPin6} />
          <PinsDigitalItem pinNumber={7} isEnabled={digitalPin7} />
          <PinsDigitalItem pinNumber={8} isEnabled={digitalPin8} />
          <PinsDigitalItem pinNumber={9} isEnabled={digitalPin9} />
          <PinsDigitalItem pinNumber={10} isEnabled={digitalPin10} />
          <PinsDigitalItem pinNumber={11} isEnabled={digitalPin11} />
          <PinsDigitalItem pinNumber={12} isEnabled={digitalPin12} />
          <PinsDigitalItem pinNumber={13} isEnabled={digitalPin13} />
        </div>
      </div>
    </div>
  )
}

const styles = {
  container: {
    height: "26px",
    overflowY: "hidden",
  },
  noScrollbar: {
    overflowX: "scroll",
    overflowY: "hidden",
    outline: "none",
    height: "80px",
  },
  wrapper: {
    float: "left",
    width: "450px",
  },
}

export default PinsDigitalBar
