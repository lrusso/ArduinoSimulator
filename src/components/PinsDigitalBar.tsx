import React from "react"
import PinsDigitalLabel from "./PinsDigitalLabel"
import PinsDigitalItem from "./PinsDigitalItem"
import { useSimulatorContext } from "../contexts/SimulatorContext"
import { isMega } from "../../src/utils/service"

const PinsDigitalBar = () => {
  const { digitalPins, boardType } = useSimulatorContext()
  const isMegaBoard = isMega(boardType)

  return (
    <div style={styles.container}>
      <div style={styles.noScrollbar}>
        <div
          style={{
            width: isMegaBoard ? "1250px" : "550px",
            ...styles.wrapper,
          }}
        >
          <PinsDigitalLabel />
          {digitalPins.slice(0, 14).map((pin, index) => (
            <PinsDigitalItem key={index} gpio={pin} />
          ))}
          {isMegaBoard && (
            <>
              {digitalPins.slice(14, 54).map((pin, index) => (
                <PinsDigitalItem key={index} gpio={pin} />
              ))}
            </>
          )}
        </div>
      </div>
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    height: "26px",
    overflowY: "hidden",
  },
  noScrollbar: {
    overflowX: "scroll",
    overflowY: "hidden",
    outline: "none",
    height: "80px",
  },
  wrapper: {
    float: "left",
  },
}

export default PinsDigitalBar
