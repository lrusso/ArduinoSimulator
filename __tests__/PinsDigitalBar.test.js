import React from "react"
import renderer from "react-test-renderer"
import PinsDigitalBar from "../src/components/PinsDigitalBar"

describe("PinsDigitalBar", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<PinsDigitalBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
