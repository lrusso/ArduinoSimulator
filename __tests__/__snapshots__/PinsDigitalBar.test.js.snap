// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinsDigitalBar Rendering 1 1`] = `
<div
  style={
    {
      "height": "26px",
      "overflowY": "hidden",
    }
  }
>
  <div
    style={
      {
        "height": "80px",
        "outline": "none",
        "overflowX": "scroll",
        "overflowY": "hidden",
      }
    }
  >
    <div
      style={
        {
          "float": "left",
          "width": "550px",
        }
      }
    >
      <div
        style={
          {
            "cursor": "default",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "13px",
            "fontWeight": "bold",
            "lineHeight": 2,
            "marginLeft": "7px",
            "marginRight": "1px",
          }
        }
      >
        DIGITAL PINS
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        0
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        1
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        2
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        3
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        4
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        5
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        6
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        7
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        8
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        9
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        10
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        11
      </div>
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": "16px",
            "color": "white",
            "cursor": "default",
            "display": "flex",
            "float": "left",
            "fontFamily": "Arial",
            "fontSize": "11px",
            "height": "16px",
            "justifyContent": "center",
            "marginLeft": "4px",
            "marginTop": "4px",
            "width": "16px",
          }
        }
      >
        12
      </div>
    </div>
  </div>
</div>
`;
